tests/test_pipeline_deploy.py

```diff
@@ -0,0 +1,5 @@
+#################################################################
+#   Libraries
+#################################################################
+import sys, os
+import pytest
```

Comment:
I'd support unit testing some of the pipeline deploy functions, and probably breaking some of those up into smaller chunks to unit test more easily.